### PR TITLE
chore: Consolidating form experience (Bootstrap to AntD) - iteration 4

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Migrates `Form` components from Bootstrap to AntD (iteration 4).

See: #10254

@rusackas @junlincc @pkdotson

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screen Shot 2021-04-28 at 2 47 10 PM
Screen Shot 2021-04-28 at 2 51 28 PM
Screen Shot 2021-04-28 at 3 47 26 PM
Screen Shot 2021-04-28 at 4 19 51 PM
Screen Shot 2021-04-28 at 5 58 35 PM
Screen Shot 2021-04-28 at 6 12 36 PM
Screen Shot 2021-04-28 at 6 14 04 PM
Screen Shot 2021-04-28 at 6 42 11 PM

### TEST PLAN
1 - Navigate to the following areas:
- Explore > Add filter > Custom SQL
- Datasets -> Edit Dataset
- Charts -> Edit Chart
- Explore -> Change Dataset

2 - Check that the layouts and functionality are working

### ADDITIONAL INFORMATION
- [] Has associated issue:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by michael-s-molina** [comment]
> thanks for the PR and enhancing the product experience! for QA purpose, could we list ALL areas that needs to be tested in the description? thanks

Done 😉 

TEST PLAN
1 - Navigate to the following areas:

Explore > Add filter > Custom SQL
Datasets -> Edit Dataset
Charts -> Edit Chart
Explore -> Change Dataset

2 - Check that the layouts and functionality are working

**Review by junlincc — COMMENTED** [review]
Does Metric use the same `form` or is it not related ? It doesn't look right. 
Screen Shot 2021-05-11 at 11 00 44 AM

**Comment by michael-s-molina** [comment]
> Does Metric use the same `form` or is it not related ? It doesn't look right.
> Screen Shot 2021-05-11 at 11 00 44 AM

@junlincc This component was not touched by this PR. I can also confirm that your reported problem is also happening on master so it's probably another issue.

File: superset-frontend/src/datasource/ChangeDatasourceModal.tsx [hunk 1]
```diff
@@ -107,7 +107,7 @@ const ChangeDatasourceModal: FunctionComponent<ChangeDatasourceModalProps> = ({
   const [filter, setFilter] = useState<any>(undefined);
   const [confirmChange, setConfirmChange] = useState(false);
   const [confirmedDataset, setConfirmedDataset] = useState<Datasource>();
-  let searchRef = useRef<HTMLInputElement>(null);
+  const searchRef = useRef<any>(null);
```
[the remark this answers is not in the document]
Comment (michael-s-molina) [reply superset-frontend/src/datasource/ChangeDatasourceModal.tsx]: Fixed.

**Comment by junlincc** [comment]
you are right, it's also happening on master. i'm surprised that Metrics and Filter don't share the same component. otherwise lgtm, thanks again!

**Review by rusackas — APPROVED** [review]
Code LGTM! Die, react-bootstrap, die! :)

**Pull request merged**